File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/5.java
public class AFDInicioFimIgual {
    public static boolean inicioFimIgual(String s) {
        if (s.length() < 1) return false;
        return s.charAt(0) == s.charAt(s.length() - 1);
    }

    public static void main(String[] args) {
        System.out.println(inicioFimIgual("101"));  // True
        System.out.println(inicioFimIgual("100"));  // False
    }
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/5.go
package main
import "fmt"

func inicioFimIgual(s string) bool {
    if len(s) < 1 {
        return false
    }
    return s[0] == s[len(s)-1]
}

func main() {
    fmt.Println(inicioFimIgual("101"))  // True
    fmt.Println(inicioFimIgual("100"))  // False
}
